internal/controller/certificates/policies/checks.go

```diff
@@ -94,6 +94,39 @@ func SecretPrivateKeyMatchesSpec(input Input) (string, string, bool) {
 	return "", "", false
 }
 
+// SecretKeystoreFormatMatchesSpec - When the keystore is not defined, the keystore
+// related fields are removed from the secret.
+// When one or more key stores are defined, re-issuance ensure that the
+// corresponding secrets are generated.
+// If the private key rotation is set to "Never", the key store related values are re-encoded
+// as per the certificate specification
+func SecretKeystoreFormatMatchesSpec(input Input) (string, string, bool) {
+	if input.Certificate.Spec.Keystores == nil {
+		if len(input.Secret.Data[cmapi.Pkcs12SecretKey]) != 0 ||
+			len(input.Secret.Data[cmapi.Pkcs12TruststoreKey]) != 0 ||
+			len(input.Secret.Data[cmapi.JksSecretKey]) != 0 ||
+			len(input.Secret.Data[cmapi.JksTruststoreKey]) != 0 {
+			return SecretMismatch, "Keystore is not defined", true
+		}
+		return "", "", false
+	}
+
+	if input.Certificate.Spec.Keystores.JKS != nil {
+		if len(input.Secret.Data[cmapi.JksSecretKey]) == 0 ||
+			len(input.Secret.Data[cmapi.JksTruststoreKey]) == 0 {
+			return SecretMismatch, "JKS Keystore keys does not contain data", true
+		}
+	}
```

Comment:
You also need to account for the case where `Certificate.Spec.Keystores.JKS.Create == false`.

The following command causes cert-manager to go into an endless issuing loop:

`kubectl patch certificate -n cert-manager-test selfsigned-cert --type merge --patch '{"spec":{"keystores":{"jks": {"create": false}}}}'`

Same with the PKCS12 keystore.

Reply:
Have made the changes. Please let me know if it looks good!